hadoop-hdds/rocks-native/src/main/java/org/apache/hadoop/hdds/utils/db/managed/ManagedSSTDumpIterator.java

```diff
@@ -0,0 +1,238 @@
+   * Throws Runtime exception in the case iterator is closed or
+   * the native Dumptool exited with non zero exit value.
+   */
+  private void checkSanityOfProcess() {
+    if (!this.open.get()) {
+      throw new RuntimeException("Iterator has been closed");
+    }
+    if (sstDumpToolTask.getFuture().isDone()
+            && sstDumpToolTask.exitValue() != 0) {
+      throw new RuntimeException("Process Terminated with non zero " +
+              String.format("exit value %d", sstDumpToolTask.exitValue()));
+    }
+  }
+
+  /**
+   *
+   * @return
+   * Throws Runtime Exception in case of SST File read failure
+   */
+
+  @Override
+  public boolean hasNext() {
+    checkSanityOfProcess();
+    return nextKey != null;
+  }
+
+  /**
+   *
+   * @return next Key
+   * Throws Runtime Exception incase of failure.
+   */
+  @Override
+  public KeyValue next() {
+    lock.lock();
+    try {
+      checkSanityOfProcess();
+      currentKey = nextKey;
+      nextKey = null;
+      while (!currentMatcher.find()) {
+        try {
+          if (prevMatchEndIndex != 0) {
+            stdoutString = new StringBuilder(stdoutString.substring(
+                    prevMatchEndIndex, stdoutString.length()));
+            prevMatchEndIndex = 0;
+            currentMatcher = PATTERN_MATCHER.matcher(stdoutString);
+          }
+          Thread.sleep(pollIntervalMillis);
+          int numberOfCharsRead = processOutput.read(charBuffer);
+          if (numberOfCharsRead < 0) {
+            if (currentKey != null) {
+              currentKey.setValue(stdoutString.toString());
+            }
+            return currentKey;
+          }
+          stdoutString.append(charBuffer, 0, numberOfCharsRead);
+          currentMatcher.reset();
+        } catch (IOException | InterruptedException e) {
+          throw new RuntimeIOException(e);
+        }
+      }
+      if (currentKey != null) {
+        currentKey.setValue(stdoutString.substring(prevMatchEndIndex,
+                currentMatcher.start()));
+      }
+      prevMatchEndIndex = currentMatcher.end();
+      nextKey =  new KeyValue(
+              currentMatcher.group(PATTERN_KEY_GROUP_NUMBER),
+              currentMatcher.group(PATTERN_SEQ_GROUP_NUMBER),
+              currentMatcher.group(PATTERN_TYPE_GROUP_NUMBER));
+      return currentKey;
+    } finally {
+      lock.unlock();
+    }
+  }
+
+  @Override
+  public synchronized void close() throws Exception {
+    lock.lock();
+    try {
+      if (this.sstDumpToolTask != null) {
+        if (!this.sstDumpToolTask.getFuture().isDone()) {
+          this.sstDumpToolTask.getFuture().cancel(true);
+        }
+        this.processOutput.close();
+      }
+      open.compareAndSet(true, false);
+    } finally {
+      lock.unlock();
+    }
+  }
+
+  @Override
+  protected void finalize() throws Throwable {
+    this.close();
+  }
+
+  /**
+   * Class containing Parsed KeyValue Record from Sst Dumptool output.
+   */
+  public static final class KeyValue {
```

Comment:
Does this need to be `public`?

Reply:
yes, since this would be used in the snapdiff code.

Reply:
Should I move this to another class file?

Reply:
Yup you can move it to another class with a not-so-generic name like `SstDumpKVEntry`.

Reply:
Yeah will move this in the snapdiff PR